editor/src/ui/language-switcher.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "./shadcn/ui/select";
+
+const languages = [
+	{ code: "en", name: "English" },
+	{ code: "fr", name: "Français" },
+	{ code: "ja", name: "日本語" },
+	{ code: "zh", name: "中文" },
+];
+
+export interface ILanguageSwitcherProps {
+	/**
+	 * Optional className for styling
+	 */
+	className?: string;
+}
+
+/**
+ * Language switcher component that allows users to change the application language.
+ */
+export function LanguageSwitcher(props: ILanguageSwitcherProps) {
+	const { i18n } = useTranslation();
+	const [currentLanguage, setCurrentLanguage] = useState(i18n.language);
+
+	const handleLanguageChange = (language: string) => {
+		i18n.changeLanguage(language);
```

Comment:
I understand this should update the entire editor app, which is great. I'm ok to require a restart/refresh of the editor when the language changed.

Reply:
Also please save the current language in local storage